### ISSUE TITLE
[keba] Communication with KEBA P30 x-series not working

### ISSUE THREAD

**Issue by AJ1979** [openhab/openhab-addons#3880]
Items on sitemap not updated, the communication with Wallbox is not working as expected

## Expected Behavior

The keba items present in the sitemap should be updated

## Current Behavior

The communication between openhab and Keba wallbox is not working.
The Wallbox itself is configured properly and has been proven by sending UDP request from a laptop to the wallbox, the wallbox sends the expected answers back to the laptop.

```
2018-09-01 23:05:02.789 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception
java.lang.NullPointerException: null
	at org.openhab.binding.keba.handler.KeContactTransceiver.stop(KeContactTransceiver.java:122) ~[?:?]
	at org.openhab.binding.keba.handler.KeContactTransceiver.unRegisterHandler(KeContactTransceiver.java:173) ~[?:?]
	at org.openhab.binding.keba.handler.KeContactHandler.lambda$0(KeContactHandler.java:150) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2018-09-01 23:05:02.868 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception
java.lang.NullPointerException: null
	at org.openhab.binding.keba.handler.KeContactTransceiver.stop(KeContactTransceiver.java:122) ~[?:?]
	at org.openhab.binding.keba.handler.KeContactTransceiver.unRegisterHandler(KeContactTransceiver.java:173) ~[?:?]
	at org.openhab.binding.keba.handler.KeContactHandler.lambda$0(KeContactHandler.java:150) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

[Archiv.zip](https://github.com/openhab/openhab2-addons/files/2342814/Archiv.zip)

## Your Environment

* Version used: openHAB 2.3.0-1, binding-keba - 2.3.0
* Operating System and version: openHABian on raspberry pi

**Comment by AJ1979** [comment]
UDP communication via sendip is working as expected between RaspberryPi and Keba P30:
[00:27:47] openhabian@openHABianPi:/etc/openhab2/items$ sudo sendip -p ipv4 -is 10.0.0.4 -p udp -us 7090 -ud 7090 -d "i" -v 10.0.0.43
Added 26 options
Initializing module ipv4
Initializing module udp
Finalizing module udp
Finalizing module ipv4
Final packet data:
45 00 00 1D   E...
1F A7 00 00   ....
FF 11 87 FA   ....
0A 00 00 04   ....
0A 00 00 2B   ...+
1B B2 1B B2   ....
00 09 4B 49   ..KI
69   i
Sent 29 bytes to 10.0.0.43
Freeing module ipv4
Freeing module udp


related tcpdump:
[00:28:37] openhabian@openHABianPi:~$ sudo tcpdump udp port 7090 -vv -X
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
00:28:50.594972 IP (tos 0x0, ttl 255, id 8103, offset 0, flags [none], proto UDP (17), length 29)
    openHABianPi.7090 > 10.0.0.43.7090: [udp sum ok] UDP, length 1
	0x0000:  4500 001d 1fa7 0000 ff11 87fa 0a00 0004  E...............
	0x0010:  0a00 002b 1bb2 1bb2 0009 4b49 69         ...+......KIi
00:28:50.601831 IP (tos 0x0, ttl 254, id 32254, offset 0, flags [none], proto UDP (17), length 71)
    10.0.0.43.7090 > openHABianPi.7090: [udp sum ok] UDP, length 43
	0x0000:  4500 0047 7dfe 0000 fe11 2a79 0a00 002b  E..G}.....*y...+
	0x0010:  0a00 0004 1bb2 1bb2 0033 885d 2246 6972  .........3.]"Fir
	0x0020:  6d77 6172 6522 3a22 5033 3020 7620 332e  mware":"P30.v.3.
	0x0030:  392e 3130 2028 3137 3131 3133 2d31 3033  9.10.(171113-103
	0x0040:  3033 3029 222c 0a                        030)",.

**Comment by AJ1979** [comment]
KEBA P30 x-series firmware version: "Firmware":"P30 v 3.9.10 (171113-103030)",\n

Just noticed that in the KebaBindingConstants.java the used KEBA firmware is not listet, but I don't know if this is the root cause:
------
public enum KebaFirmware {
        V201M21("2.01m21"),
        V22A1("2.2a1"),
        V23A2("2.3a2"),
        V23A3("2.3a3"),
        V25A3("2.5a3"),
        V3042A1("3.04.2a1"),
        V3062A5("3.06.2a5"),
        V3071A1("3.07.1a1"),
        V3084("3.08.4");
------

**Comment by tiptick** [comment]
i think this is fixed in #4987 , the binding is not longer dependent on FIrmware versions.

**Comment by kaikreuzer** [comment]
Correct.
